mypy/plugins/common.py

```diff
@@ -141,7 +141,7 @@ def find_shallow_matching_overload_item(overload: Overloaded, call: CallExpr) ->
                         break
                 elif (
                     arg_none
-                    and not is_optional(arg_type)
+                    and not can_be_none(arg_type)
```

Comment:
It seems to me that this logic should also use the new `can_be_none()`, but I'm not sure how to test this code path.